### DISCUSSION TITLE
Cylindrical source

### DISCUSSION THREAD

**Discussion by fnovais11** [fusion-energy/neutronics-workshop#118] [Q&A]
Hello.
Is there an easy way to make an infinite cylindrical source in openmc? I am assuming it would be similar to your ring source, but I am having a hard time trying to modify it.

I have a geometry like this:
c1=openmc.ZCylinder(x0=0.0, y0=0.0, r=300.0, boundary_type="reflective")
c2=openmc.ZCylinder(x0=0.0, y0=0.0, r=100.0)

breeding_zone = +c2 & - c1
cell_bz=openmc.Cell(region = breeding_zone)
cell_bz.fill = mixed_mat

I want to place an axial source inside.

Thanks,
Felipe.

**Comment by shimwell** [comment]
Interesting source.

From the docs I can see that the Z arg accepts univariates which might do what you are after
 https://docs.openmc.org/en/stable/pythonapi/generated/openmc.stats.CylindricalIndependent.html#openmc.stats.CylindricalIndependent

It might also be worth plotting the OpenMC source with https://github.com/fusion-energy/openmc_source_plotter

**Reply by fnovais11** [reply]
I am trying to modify the ring source example:

`z_values = openmc.stats.Discrete([0], [1])
angle = openmc.stats.Uniform(a=0.0, b=2 * 3.14159265359)
my_source.space = openmc.stats.CylindricalIndependent(.....`

My confusion is what `z_values = openmc.stats.Discrete([0], [1]` actually represents.

I have also been trying to use `openmc_source_plotter` but it fails and kills my kernel.

**Comment by shimwell** [comment]
The ```openmc.stats.Discrete([0], [1])``` sets the z value to 0 for 100% of the particles

Setting ```openmc.stats.Discrete([42], [1])``` sets the z value to 42cm for 100% of the particles.

Lists are also acceptable so you could do ```[10,20], [0.5, 0.5]``` for 50% of the particles to be at z=10 and 50% at z=20

However you want a different Univariate to the Discrete Univariate. I think you want a ```Uniform``` distribution of Z, R and phi values

Perhaps something like this is what you are after

```python
openmc.stats.CylindricalIndependent(
      r=openmc.stats.Uniform(a=0, 100),  # 0 cm to 100 cm radius of cylinder
      phi=openmc.stats.Uniform(a=0, b=2*3.141592653589793238),  # 0 radians to 2 pi radians
      z=openmc.stats.Uniform(a=-10, b=10), # -10 cm to 10cm in height
      origin=(0.0, 0.0, 0.0),
)
```

For the openmc_source_plotter it just got updated a few days ago but it does need the latest development version of openmc to work. If it still crashes with the new versions of both packages would you mind sending me any error messages. Thanks
